app.rb

```diff
@@ -1,20 +1,236 @@
+  #   data[:apartments].each do |ap|
+  #     puts "The following people live in apartment number " + ap[:id].to_s
+  #     data[:tenants].each_index.select{|i| data[:tenants][i][:apartment_id] == ap[:id]}.each {|t| puts data[:tenants][t][:name]}
+  #     puts "********"
+  #   end
+
+#-------------------------Part 2-------------------------------
+
+    puts "
+          Welcome to Tenant / Apartment Management System
+          Input following commands to proceed
+          "
+      userInput=nil
+
+      userInput = 'menu'
+        while !(userInput==nil)
+          case userInput
+          when "1"
+             ruby_apartments.each {|ap| puts "Property:# #{ap.id}; Located at: #{ap.address}"}
+             userInput = gets.chomp
+          when "2"
+             ruby_tenants.each {|t| puts "Tenant ID: #{t.id}; Tenant Name: #{t.name};"}
+             userInput = gets.chomp
+          when "3"
+             ruby_apartments.each do |ap|
+               puts "The following people live in apartment number #{ap.id}."
+               ruby_tenants.each_index.select{|i| ruby_tenants[i].apartment_id == ap.id}.each {|t| puts ruby_tenants[t].name}
+               puts "------------------------"
+             end
+             userInput = gets.chomp
+          when "4"
+            ruby_tenants.each {|t| puts "Tenant ID: #{t.id}; Tenant Name:'#{t.name}';"}
+            puts "Enter tenant 'id' to search for tenant address"
+            userInput = gets.chomp
+              apartment_id = nil
+              apartment_id = ruby_tenants.find{|ap| ap.id == userInput.to_i}.apartment_id
+              puts "Tenants address: #{ruby_apartments.find{|ap| ap.id == apartment_id}.address}"
+            userInput = gets.chomp
+          when "5"
+            tenantManipulator=[];
+            # Print list of apartments with IDs and Addresses
+            ruby_apartments.each {|ap| puts "Apartment ID: #{ap.id}; Apartment Address:#{ap.address};"}
+            # User inputs apartment selection
+            puts "Enter apartment 'id' to search for apartment occupancy list"
+            userInput = gets.chomp
+            #Output apartment's occupants list
+            occupants = ruby_apartments.find{|ap| ap.id == userInput.to_i}.occupants
+            puts "Apartment number #{userInput.to_i} has #{occupants.size} occupant(s), detailed list is included below "
+            ruby_apartments.find{|ap| ap.id == userInput.to_i}.occupants.each_with_index {|occupant, index| puts "#{index+1} . #{occupant[:name]}"}
+            userInput = gets.chomp
+          when "6"
+            # print tenant array as numbered list
+
+            # ask for action.. add remove transfer
+            puts "Type 'Add', 'Delete', or 'Transfer' to manage Tenant Lists" #Otherwise, pick a number (1-7), type in 'menu' for a list of options, or 'exit' to exit."
+            userInput = gets.chomp
+
+            if userInput.downcase =~ /add/
```

Comment:
I was wondering why you decided on regular expression here instead of the string. Regular expressions are a really strong and powerful tool, but based on the code here, a string will do the job just as well. If you were looking for a pattern in the user's input that be something different. Ie anything that looks like add -  `/add\w*/`  This regex will return true for things like `addition` `addend` `addibaba` basically the string has to have add as the first 3 letters and then can be follow by 0 or more word characters.